@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,6 +256,89 @@ export namespace MapViewUtils {
         return result;
     }
 
+    /**
+     * Returns the height of the camera above the earths surface.
+     *
+     * If there is an ElevationProvider, this is used. Otherwise the projection is used to determine
+     * how high the camera is above the surface.
+     *
+     * @param level Which level to request the surface height from.
+     * @return Height in world units.
+     */
+    function getHeightAboveTerrain(
+        location: THREE.Vector3,
+        projection: Projection,
+        elevationProvider?: ElevationProvider,
+        level?: number
+    ): number {
+        if (elevationProvider !== undefined) {
+            const geoLocation = projection.unprojectPoint(location);
+            const heightAboveTerrain = elevationProvider.getHeight(geoLocation, level);
+            if (heightAboveTerrain !== undefined) {
+                const height = projection.unprojectAltitude(location) - heightAboveTerrain;
+                return Math.max(height, 1);
+            }
+        }
+        return Math.abs(projection.groundDistance(location));
+    }
+
+    /**
+     * @internal
+     * @deprecated This method will be moved to MapView.
+     */
+    export function getTargetAndDistance(
+        projection: Projection,
+        camera: THREE.PerspectiveCamera,
+        elevationProvider?: ElevationProvider
+    ): { target: THREE.Vector3; distance: number } {
+        const cameraPitch = extractAttitude({ projection }, camera).pitch;
```

Comment:
We just need to remember about improving pitch calculus - now is doing too much....

Reply:
yeah, @zbigg also commented in my other PR, so I will remember :-)